src/Compilers/CSharp/Portable/Compilation/AwaitExpressionInfo.cs

```diff
@@ -37,15 +31,12 @@ public override bool Equals(object obj)
 
         public bool Equals(AwaitExpressionInfo other)
         {
-            return object.Equals(this.GetAwaiterMethod, other.GetAwaiterMethod)
-                && object.Equals(this.IsCompletedProperty, other.IsCompletedProperty)
-                && object.Equals(this.GetResultMethod, other.GetResultMethod)
-                && this.IsDynamic == other.IsDynamic;
+            return _awaitableInfo.Equals(other);
```

Comment:
>_awaitableInfo.Equals(other); [](start = 19, length = 29)

It looks like this is always going to return false. If this is the case, it looks like we have a test gap (nothing failed), it will be good to close it.